mono/metadata/marshal.c

```diff
@@ -276,6 +276,21 @@ mono_marshal_init_tls (void)
 static MonoObject*
 mono_object_isinst_icall (MonoObject *obj, MonoClass *klass)
 {
+	if (!klass)
+		return NULL;
+
+	/* This is called from stelemref so it is expected to succeed */
+	/* Fastpath */
+	if (mono_class_is_interface (klass)) {
+		MonoVTable *vt = obj->vtable;
+
+		if (!klass->inited)
```

Comment:
This is called from the 'complex' stelemref wrapper, and its just a fastpath, if the check fails, we still need to do all the other checks.